isis/src/control/objs/ControlNet/ControlNet.cpp

```diff
@@ -369,9 +373,16 @@ namespace Isis {
     // make sure there is a node for every measure
     for (int i = 0; i < point->GetNumMeasures(); i++) {
       QString sn = point->GetMeasure(i)->GetCubeSerialNumber();
-      if (!cubeGraphNodes->contains(sn)) {
+       if (!cubeGraphNodes->contains(sn)) {
         cubeGraphNodes->insert(sn, new ControlCubeGraphNode(sn));
       }
+      // if the graph doesn't have the sn
+      if (!m_vertexMap.contains(sn)) {
+        ImageVertex newVertex = boost::add_vertex(m_controlGraph); 
+        m_controlGraph[newVertex].serial = sn; 
+        m_vertexMap.insert(sn, newVertex);
+  //    boost::put(indexMapAdaptor, newVertex, i); <-- what does this do? 
```

Comment:
This basically assigns numbers to the vertices in the graph. The connected components algorithm needs this map in order to work when vertices are stored in a list or set. We may want to recreate this map everytime we call the connected components method instead of trying to store it and keep it up to date.

Reply:
See the vertex_index_map parameter https://www.boost.org/doc/libs/1_67_0/libs/graph/doc/connected_components.html